mockapi/mock_elasticache_api.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Mocks for service API. For unit tests.
+
+package mockapi
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	svcsdk "github.com/aws/aws-sdk-go/service/elasticache"
+	svcsdkapi "github.com/aws/aws-sdk-go/service/elasticache/elasticacheiface"
+)
+
+// Allows mocking ElastiCache API methods.
+type MockElastiCacheAPI struct {
+	svcsdkapi.ElastiCacheAPI
+
+	// mock api outputs
+	MockDecreaseReplicaCountOutput *svcsdk.DecreaseReplicaCountOutput
+	MockDecreaseReplicaCountError error
+	MockIncreaseReplicaCountOutput *svcsdk.IncreaseReplicaCountOutput
+	MockIncreaseReplicaCountError error
+	// ...
+}
+
+func (m MockElastiCacheAPI) DecreaseReplicaCountWithContext(
+	aws.Context,
+	*svcsdk.DecreaseReplicaCountInput,
+	...request.Option,
+) (*svcsdk.DecreaseReplicaCountOutput, error) {
+	return m.MockDecreaseReplicaCountOutput, m.MockDecreaseReplicaCountError
+}
+
+func (m MockElastiCacheAPI) IncreaseReplicaCountWithContext(
+	aws.Context,
+	*svcsdk.IncreaseReplicaCountInput,
+	...request.Option,
+) (*svcsdk.IncreaseReplicaCountOutput, error) {
+	return m.MockIncreaseReplicaCountOutput, m.MockIncreaseReplicaCountError
+}
+
+// add more API methods implementation to MockElastiCacheAPI as needed for unit testing
```

Comment:
Instead of doing this, how about use the `mockery` tool to automatically generate a mocked ElasticacheAPI struct?

Reply:
Thanks for your inputs @jaypipes 
I have updated the code to use `mockery`.